test/functional/kernel/reduce-basic/test-kernel-loc.cpp.in

```diff
@@ -0,0 +1,218 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+//
+// test/include headers
+//
+#include "RAJA_test-base.hpp"
+#include "RAJA_test-camp.hpp"
+#include "RAJA_test-index-types.hpp"
+
+// for data types
+#include "RAJA_test-reduce-types.hpp"
+#include "RAJA_test-reduceloc-types.hpp"
+#include "RAJA_test-reduceloc-data.hpp"
+
+//
+// Header for tests in ./tests directory
+//
+// Note: CMake adds ./tests as an include dir for these tests.
+//
+#include "test-kernel-loc-@LOC_TYPE@.hpp"
+
+
+//
+// Exec pols for kernel region tests
+//
+
+using SequentialKernelLocExecPols =
+  camp::list<
+
+    RAJA::KernelPolicy<
+      RAJA::statement::For<1, RAJA::loop_exec,  // row
+        RAJA::statement::For<0, RAJA::loop_exec,  // col
+          RAJA::statement::Lambda<0>
+        >
+      >
+    >,
+
+    RAJA::KernelPolicy<
+      RAJA::statement::For<1, RAJA::seq_exec,  // row
+        RAJA::statement::For<0, RAJA::seq_exec,  // col
+          RAJA::statement::Lambda<0>
+        >
+      >
+    >
+
+  >;
+
+using SequentialKernelLocReducePols =
+  camp::list<
+    RAJA::seq_reduce
+  >;
+
+using SequentialKernelLocForallPols =
+  camp::list<
+    RAJA::seq_exec
+  >;
+
+#if defined(RAJA_ENABLE_OPENMP)
+
+using OpenMPKernelLocExecPols =
+  camp::list<
+
+    RAJA::KernelPolicy<
+      RAJA::statement::For<1, RAJA::omp_parallel_for_exec,  // row
+        RAJA::statement::For<0, RAJA::omp_parallel_for_exec,  // col
```

Comment:
Do we want to have nested parallelism here?

Reply:
It doesn't fail when run, but I can make the outer loop a RAJA::seq_exec.

Reply:
or you could use a loop collapse policy....maybe not the best for performance, but it would be a good test.

Reply:
@MrBurmark @rhornung67 Modified these policies for use of collapse and seq_exec.

Reply:
I'm happy with that, should we add collapse to omp target?

Reply:
There doesn't appear to be a corresponding collapse policy for omp_target.

Reply:
Yes, there is. It's used in many kernels in the RAJA Perf Suite; e.g., https://github.com/LLNL/RAJAPerf/blob/develop/src/basic/NESTED_INIT-OMPTarget.cpp#L67

Reply:
Added it in now. Thanks!

Reply:
@rhornung67 I've commented out the omp_target_parallel_collapse_exec test for the moment. It always produces the wrong answer, most likely because the data is not being copied over to the device. Does this work correctly in the PerfSuite?

Reply:
All of the PerfSuite kernels that use the omp target collapse stuff (RAJA::kernel) yield correct results. However, none of those kernels use reductions. There are several kernels that use reductions with RAJA::forall and those produce correct results for omp target. There may be a missing 'map' clause in the code someplace.